src/app/qgslayerstylingwidget.cpp

```diff
@@ -743,6 +743,7 @@ void QgsLayerStylingWidget::setAnnotationItem( QgsAnnotationLayer *layer, const
   mContext.setAnnotationId( itemId );
   if ( layer )
     setLayer( layer );
+  mStackedWidget->setCurrentIndex(1);
```

Comment:
Shouldn't this always be triggered by the setLayer command above? What's the sequence of events leading to layer being nullptr but a valid annotation item being selected?

Reply:
> Shouldn't this always be triggered by the setLayer command above?

the `setLayer` function often does not reach `mStackedWidget->setCurrentIndex( 1 );` Consider the following sequence:

- layer styling widget is created, `mCurrentLayer` is set to the main annotation layer (the only layer in a newly opened project)
- now create an annotation item 
- `setAnnotationItem` is called which also calls `setLayer`, but `setLayer` immediately returns because layer equals `mCurrentLayer` (both reference main annotation layer)
-> this results in the fact that you could not edit annotation items of the main annotation layer, because it would always stay on the 'notSupportedPage'

-- 
But I agree that `mStackedWidget->setCurrentIndex( 1 );` should probably be within the if-statement. It does not make sense to show the layer page if layer is null.